packages/@lwc/engine-core/src/framework/renderer.ts

```diff
@@ -65,9 +65,10 @@ export interface RendererAPI {
     createCustomElement: (
         tagName: string,
         upgradeCallback: LifecycleCallback,
-        useNativeLifecycle: boolean
+        useNativeLifecycle: boolean,
+        isFormAssociated: boolean | undefined
     ) => E;
-    defineCustomElement: (tagName: string) => void;
+    defineCustomElement: (tagName: string, isFormAssociated: boolean | undefined) => void;
```

Comment:
For these `renderer` APIs, I would prefer to be explicit and just accept a `boolean`, not `boolean | undefined`, but I also don't think this is a big deal.

Reply:
@nolanlawson I updated the `renderer` APIs to only accept `boolean` and now normalize the `formAssociated` value to explicitly be a `boolean`.

Technically component authors can pass any value to `formAssociated` but since we're using the `UpgradableConstructor`, multiple constructors can be shared to a single entry in the custom element registry I think the value should be a strict `boolean`.